crates/red_knot_python_semantic/src/types.rs

```diff
@@ -445,12 +445,30 @@ impl<'db> Type<'db> {
     }
 
     #[must_use]
-    pub fn to_instance(&self) -> Type<'db> {
+    pub fn to_instance(&self, db: &'db dyn Db) -> Type<'db> {
         match self {
             Type::Any => Type::Any,
             Type::Unknown => Type::Unknown,
+            Type::Never => Type::Never,
             Type::Class(class) => Type::Instance(*class),
-            _ => Type::Unknown, // TODO type errors
+            Type::Union(union) => union.map(db, |element| element.to_instance(db)),
+            // TODO: we can probably do better here: --Alex
+            Type::Intersection(_) => Type::Unknown,
+            // TODO: converting to `Unknown` here is probably correct,
+            // but should result in a diagnostic reporting the use of an unbound name:
```

Comment:
This diagnostic should be handled by just visiting all Name exprs with Load context and checking if their type is unbound, it shouldn't be handled in every method of `Type` where we see an `Unbound`. So I'm not sure we need a TODO for it here.

(TBH I'm getting more and more convinced that `Unbound` should not be a `Type` variant at all.)